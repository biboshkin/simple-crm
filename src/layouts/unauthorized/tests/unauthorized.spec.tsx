import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import { UnauthorizedLayout } from "..";

test("render correct", () => {
  const component = renderer.create(
    <UnauthorizedLayout title="Заголовок" description="Описание">
      <div />
    </UnauthorizedLayout>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test("render title optionally", () => {
  const wrapper = shallow(
    <UnauthorizedLayout title="Заголовок">
      <div />
    </UnauthorizedLayout>
  );

  expect(wrapper.find(".title").length).toBe(1);

  const wrapperWithoutTitle = shallow(
    <UnauthorizedLayout>
      <div />
    </UnauthorizedLayout>
  );

  expect(wrapperWithoutTitle.find(".title").length).toBe(0);
});

test("render description optionally", () => {
  const wrapper = shallow(
    <UnauthorizedLayout description="Описание">
      <div />
    </UnauthorizedLayout>
  );

  expect(wrapper.find(".description").length).toBe(1);

  const wrapperWithoutTitle = shallow(
    <UnauthorizedLayout>
      <div />
    </UnauthorizedLayout>
  );

  expect(wrapperWithoutTitle.find(".description").length).toBe(0);
});
