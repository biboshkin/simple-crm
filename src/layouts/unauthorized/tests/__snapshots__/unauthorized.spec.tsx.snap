// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correct 1`] = `
<form
  className="form"
>
  <div
    className="container"
  >
    <div
      className="layout"
    >
      <div
        className="imgContainer"
      >
        <img
          alt="logo"
          className="img"
          src="logo.png"
        />
      </div>
      <h2
        className="ant-typography title"
        style={
          Object {
            "WebkitLineClamp": null,
          }
        }
      >
        Заголовок
      </h2>
      <h4
        className="ant-typography description ant-typography-secondary"
        style={
          Object {
            "WebkitLineClamp": null,
          }
        }
      >
        Описание
      </h4>
    </div>
    <div />
  </div>
</form>
`;
